sklearn/_build_utils/openmp_helpers.py

```diff
@@ -8,9 +8,6 @@
 import sys
 import textwrap
 import warnings
-import subprocess
-
-from distutils.errors import CompileError, LinkError
```

Comment:
`setuptools` introduced CompileError and LinkError, but that requires 61.1. Even the latest version of [Ubuntu LTS](https://packages.ubuntu.com/jammy/python3-setuptools) (22.04LTS) only ships with 59.6.

In our case, I think we can get away with catching all Exceptions and display the error message.